src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/Triggers/DataBoxEdgeTriggerNewCmdletBase.cs

```diff
@@ -0,0 +1,250 @@
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Net;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.EdgeGateway;
+using Microsoft.Azure.Management.EdgeGateway.Models;
+using Microsoft.Rest.Azure;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using PSResourceModel = Microsoft.Azure.PowerShell.Cmdlets.DataBoxEdge.Models.PSDataBoxEdgeTrigger;
+using ResourceModel = Microsoft.Azure.Management.EdgeGateway.Models.Trigger;
+
+
+namespace Microsoft.Azure.PowerShell.Cmdlets.DataBoxEdge.Common.Cmdlets.Triggers
+{
+    [Cmdlet(VerbsCommon.New, Constants.Trigger,
+         DefaultParameterSetName = FileEventTriggerParameterSet,
+         SupportsShouldProcess = true
+     ),
+     OutputType(typeof(PSResourceModel))]
+    public class DataBoxEdgeTriggerNewCmdletBase : AzureDataBoxEdgeCmdletBase
+    {
+        private const string FileEventTriggerParameterSet = "FileEventTriggerParameterSet";
+        private const string FileEventTriggerResourceIdParameterSet = "FileEventTriggerResourceIdParameterSet";
+
+        private const string PeriodicTimerTriggerParameterSet = "PeriodicTimerTriggerParameterSet";
+        private const string PeriodicTimerTriggerResourceIdParameterSet = "PeriodicTimerTriggerResourceIdParameterSet";
+
+        [Parameter(Mandatory = true, ParameterSetName = PeriodicTimerTriggerParameterSet,
+            HelpMessage = Constants.ResourceGroupNameHelpMessage,
+            Position = 0)]
+        [Parameter(Mandatory = true, ParameterSetName = FileEventTriggerParameterSet,
+            HelpMessage = Constants.ResourceGroupNameHelpMessage,
+            Position = 0)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = FileEventTriggerParameterSet,
+            HelpMessage = Constants.DeviceNameHelpMessage,
+            Position = 1)]
+        [Parameter(Mandatory = true, ParameterSetName = PeriodicTimerTriggerParameterSet,
+            HelpMessage = Constants.DeviceNameHelpMessage,
+            Position = 1)]
+        [ResourceNameCompleter("Microsoft.DataBoxEdge/dataBoxEdgeDevices", nameof(ResourceGroupName))]
+        [ValidateNotNullOrEmpty]
+        public string DeviceName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = FileEventTriggerParameterSet,
```

Comment:
This is conflict with the design issue: https://github.com/raviten/azure-powershell/blob/raviten/databoxedge-cmdlets-ii/src/DataBoxEdge/DataBoxEdge/help/New-AzDataBoxEdgeTrigger.md#periodictimertriggerparameterset